app/frontend/src/features/profile/queries.ts

```diff
@@ -0,0 +1,76 @@
+import api from "@/lib/api";
+import type { UserProfile } from "@server/db/users";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+
+const PROFILE_QUERY_KEY = ["profile"];
+
+// * Query options for own profile
+
+export const profileQueryOptions = {
+	queryKey: PROFILE_QUERY_KEY,
+	queryFn: async () => {
+		const response = await api.user.profile.$get();
+		if (!response.ok) {
+			throw new Error("Failed to fetch profile");
+		}
+		return response.json();
+	},
+};
+
+// * Query options for fetch user profile
+// * @param userId - The ID of the user whose profile to fetch
+
+// FIXME: Add username support like api.profile["@:username"].$get() for a better user experience
+
+export const userProfileQueryOptions = (userId: string) => ({
+	queryKey: ["userProfile", userId],
+	queryFn: async () => {
+		const response = await api.user[":id"].$get({
+			param: { id: userId },
+		});
+
+		if (!response.ok) throw new Error("Failed to fetch user profile");
+		return response.json();
+	},
+});
+
+// * Hook to fetch your own profile
+
+export function useProfile() {
+	return useQuery<UserProfile>(profileQueryOptions);
+}
+
+// * Hook to fetch user profile
+
+export function useUserProfile(userId: string) {
+	return useQuery(userProfileQueryOptions(userId));
+}
```

Comment:
There is no need to create a hook just for the queryOptions. you can just use them where you need them, by using the [queryOptions()](https://tanstack.com/query/latest/docs/framework/react/reference/queryOptions) helper function provided by tanstack query. Or look into `queries.ts` file in `src/features/chat`